Library/Formula/ruby.rb

```diff
@@ -106,14 +108,27 @@ def post_install
     config_file.write rubygems_config
 
     # Create the sitedir and vendordir that were skipped during install
-    mkdir_p `#{bin}/ruby -e 'require "rbconfig"; print RbConfig::CONFIG["sitearchdir"]'`
-    mkdir_p `#{bin}/ruby -e 'require "rbconfig"; print RbConfig::CONFIG["vendorarchdir"]'`
+    ruby="#{bin}/ruby#{program_suffix}"
+    %w[sitearchdir vendorarchdir].each do |dir|
+      mkdir_p `#{ruby} -rrbconfig -e 'print RbConfig::CONFIG["#{dir}"]'`
+    end
+
+    # Create the version-specific bindir used by rubygems
+    mkdir_p "#{rubygems_bindir}"
   end
 
   def abi_version
     "2.3.0"
   end
 
+  def program_suffix
+    build.with?("suffix") ? "23" : ""
+  end
```

Comment:
I would really like to avoid shelling out to something that was just built for such a simple piece of information. I agree that any repetition and manual information retrieval is bad, though.

> My thought is just that for stuff in core it's handy to try and reduce the stuff that needs changed on updates as much as possible.

Valid point. I'll see what I can do about this, but only after this PR is merged, if that's okay.